src/test/scala/nl/tradecloud/kafka/KafkaSubscriberSpec.scala

```diff
@@ -105,6 +106,33 @@ class KafkaSubscriberSpec extends TestKit(ActorSystem("KafkaSubscriberSpec", Con
 
       receiverProbe.expectNoMessage(defaultNegativeTimeout)
     }
+
+    "drop messages with an invalid type" in {
+      val receiverProbe = TestProbe("receiver")
+      val publisher = new KafkaPublisher(system)
+
+      val subscriber = new KafkaSubscriber(
+        serviceName = "test",
+        group = "test_group_0",
+        topics = Set("test_topic_5"),
+        system = system,
+        configurationProperties = Seq(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest")
+      )
+
+      subscriber.atLeastOnce[String](
+        Flow[KafkaMessage[String]].map { msg =>
+          receiverProbe.ref ! msg.msg
+
+          msg.offset
+        }
+      )
+
+      publisher.publish(
+        topic = "test_topic_5",
+        msg = Done
+      )
+      receiverProbe.expectNoMessage(defaultNegativeTimeout)
```

Comment:
Can we test the discarded message is committed?

Reply:
We can test if the commitScaladsl function on the CommittableOffset is called, the actual commit should be (and is) tested at the kafka streams library.